lib/CodeGen/CodeGenFunction.h

```diff
@@ -1238,6 +1238,53 @@ class CodeGenFunction : public CodeGenTypeCache {
   const TargetInfo &getTarget() const { return Target; }
   llvm::LLVMContext &getLLVMContext() { return CGM.getLLVMContext(); }
 
+public:
+  // Represents the kind of expression that the BoundsExpr in NextBoundsExprForCheck goes with
+  enum BoundsCheckKind {
+    BC_None,
+    BC_Deref,
+    BC_ArraySubscript
+  };
+
+private:
+  // We can't pass the next bounds to check against using parameters, so we save it into the
+  // current function context, along with a hint of what kind of expression we're aiming to check.
+  BoundsExpr* NextBoundsExprForCheck;
+
+  // The kind of expression that NextBoundsExprForCheck is being stored for.
+  // If this is BC_None, NextBoundsExprForCheck should definitely be nullptr.
+  BoundsCheckKind NextBoundsCheckKind;
+
+public:
+  // This retrieves the saved bounds for the next check, zeroing where they were stored
+  // This will give an assertion error if Kind does not match the previously saved kind.
+  BoundsExpr* GetAndClearNextBoundsCheckExpr(BoundsCheckKind Kind) {
+    assert(getLangOpts().CheckedC && "GetAndClearNextBoundsCheckExpr requires Checked C to be enabled");
```

Comment:
I think you can delete the message part of this assert - it is pretty clear what happened.